### PR TITLE
repositories: support listing commits

### PR DESCRIPTION
**Pull request by grahamc**
## What did you implement:

https://developer.github.com/v3/repos/commits/

1. list commits
2. get a single commit

Has a couple todos, called out with github comments. What do you think from here?

#### How did you verify your change:

```
cargo run --example repo_commits
cargo test
```

I also `cargo fmt`'d my added changes, and didn't add unrelated reformats.

#### What (if anything) would need to be called out in the CHANGELOG for the next release:

nothing, I think :)

### CODE REVIEW

File: src/repo_commits.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        RepoCommits {
+            github,
+            owner: owner.into(),
+            repo: repo.into(),
+        }
+    }
+
+    /// list repo commits
+    /// !!! make optional parameters
```
Comment (grahamc) [inline src/repo_commits.rs]: todo?

File: src/repo_commits.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    pub fn list(&self) -> Future<Vec<RepoCommit>> {
+        let uri = format!("/repos/{}/{}/commits", self.owner, self.repo);
+        self.github.get::<Vec<RepoCommit>>(&uri)
+    }
+
+    /// provides a stream over all pages of pull commits
+    /// !!! make optional parameters
```
Comment (grahamc) [inline src/repo_commits.rs]: todo?

File: src/repo_commits.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    pub fn get(&self, commit_ref: &str) -> Future<RepoCommit> {
+        let uri = format!("/repos/{}/{}/commits/{}", self.owner, self.repo, commit_ref);
+        self.github.get::<RepoCommit>(&uri)
+    }
+}
+
+// representations
+
+// !!! RepoCommit, CommitDetails, CommitRef, UserStamp are exact
+//     dupes of pull_commits.rs' representations.
```
Comment (grahamc) [inline src/repo_commits.rs]: todo?

**Comment by softprops** [comment]
Looks great!

**Pull request merged**